include/simulator.h

```diff
@@ -41,18 +42,178 @@ struct CostMetrics {
 
 class Device {
 public:
-  enum DeviceType {
-    DEVICE_GPU,
-    DEVICE_CPU,
-    DEVICE_COMM,
-  };
-  Device(DeviceType type, int node_id, int gpu_id, size_t capacity);
-  Device(DeviceType type, float bandwidth);
+    enum DeviceType {
+        DEVICE_COMP,
+        DEVICE_MEM,
+        DEVICE_COMM,
+    };
+    Device(std::string name, DeviceType type, int node_id, int socket_id, int device_id);
+    std::string name;
+    DeviceType type;
+    int node_id;
+    int socket_id;
+    int device_id;
+};
+
+class Comp_device : public Device {
 public:
-  int node_id, gpu_id;
-  float bandwidth;
-  size_t capacity;
-  DeviceType type;
+    enum CompDevType {
+        LOC_PROC,   //CPU
```

Comment:
Why are the names here unintuitive and require comments?

Reply:
Just using Legion's name for the CPU and GPU.